card/mod/bootstrap/lib/bootstrap/component.rb

```diff
@@ -91,11 +101,27 @@ def add_content content
       @content[-1] << "\n#{content}".html_safe if content.present?
     end
 
-    def process_tag tag_name
+    def process_tag tag_name, &content_block
       @content.push "".html_safe
       @append << []
-      content, opts = yield
-      add_content content
+      @wrap << []
+
+
+      content, opts = content_block.call
+      wrappers = @wrap.pop
+      if wrappers.present?
+        while wrappers.present? do
+          wrapper = wrappers.shift
+          if wrapper.is_a? Symbol
+            send wrapper, &content_block
+          else
+            instance_exec(content, &wrappers.shift)
+          end
+        end
+      else
+        add_content content
+      end
```

Comment:
The whole bootstrap builder API needs refactoring. The wrap stuff doesn't work and it's way too messy and complicated to be maintable.